src/vm/perfmap.cpp

```diff
@@ -274,6 +281,11 @@ void PerfMap::LogPreCompiledMethod(MethodDesc * pMethod, PCODE pCode)
 {
     LIMITED_METHOD_CONTRACT;
 
+    if (s_Current == nullptr)
```

Comment:
It seems that if one of these logging functions races with the PerfMap::Destroy, we have a potential to crash in the code below.

Reply:
This doesn't look like an issue with this function (since it is just an early out), but for other preexisting functions it is potentially an issue.

I'll open a separate PR to fix this.